### PR TITLE
Revert "Revert "Support upstream http filters with tcp tunneling (#27183)" (#32980)"

### PR DESCRIPTION
**Pull request by vikaschoudhary16**
…183)" (#32980)"

This reverts commit 04835afdae38f7e609ed37e489685634c83c5d1f.

Commit Message:
Additional Description:
Risk Level:
Testing:
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Fixes commit #PR or SHA]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

File: test/integration/http_protocol_integration.cc [hunk 1]
```diff
@@ -113,9 +115,11 @@ std::string HttpProtocolIntegrationTest::protocolTestParamsToString(
                       http2ImplementationToString(params.param.http2_implementation),
                       params.param.defer_processing_backedup_streams ? "WithDeferredProcessing"
                                                                      : "NoDeferredProcessing",
+                      params.param.use_universal_header_validator ? "Uhv" : "Legacy",
```
Comment (alyssawilk) [inline test/integration/http_protocol_integration.cc:118]: revert?

File: test/integration/http_protocol_integration.cc [hunk 2]
```diff
@@ -113,9 +115,11 @@ std::string HttpProtocolIntegrationTest::protocolTestParamsToString(
...
                       params.param.deprecate_callback_visitor ? "WithCallbackVisitor"
                                                               : "NoCallbackVisitor",
-                      params.param.use_universal_header_validator ? "Uhv" : "Legacy");
+                      params.param.tunneling_with_upstream_filters ? "WithUpstreamHttpFilters"
```
Comment (alyssawilk) [inline test/integration/http_protocol_integration.cc:121]: I think this is the probem.  Let's only do this for tests using TCP proxying. 

/wait

**Review by alyssawilk — COMMENTED** [review]
/wa

File: test/integration/http_protocol_integration.cc [hunk 3]
```diff
@@ -50,15 +50,17 @@ std::vector<HttpProtocolTestParams> HttpProtocolIntegrationTest::getProtocolTest
...
-        for (Http1ParserImpl http1_implementation : http1_implementations) {
-          for (Http2Impl http2_implementation : http2_implementations) {
-            for (bool defer_processing : http2_bool_values) {
-              for (bool deprecate_callback_visitor : http2_bool_values) {
-                for (bool use_header_validator : use_header_validator_values) {
-                  ret.push_back(HttpProtocolTestParams{
-                      ip_version, downstream_protocol, upstream_protocol, http1_implementation,
-                      http2_implementation, defer_processing, use_header_validator,
-                      deprecate_callback_visitor});
+        for (const bool tunneling_with_upstream_filters : {false, true}) {
```
Comment (ggreenway) [inline test/integration/http_protocol_integration.cc:53]: Please find a compromise between a reasonable amount of coverage, and doubling the number of permutations of the tests. Recent failures were partially caused by the tests taking too long due to too many permutations.

**Review by ggreenway — CHANGES_REQUESTED** [review]

**Comment by vikaschoudhary16** [comment]
thanks @alyssawilk and @ggreenway. will update

**Review by alyssawilk — APPROVED** [review]

**Review by ggreenway — APPROVED** [review]

**Pull request merged**